pkg/catalog/utils/version.go

```diff
@@ -90,3 +92,37 @@ func ReleaseServerVersion(serverVersion string) bool {
 	}
 	return true
 }
+
+func LatestAvailableTemplateVersion(template *v3.CatalogTemplate) (*v3.TemplateVersionSpec, error) {
+	versions := template.DeepCopy().Spec.Versions
+	if len(versions) == 0 {
+		return nil, errors.New("empty catalog template version list")
+	}
+
+	sort.Slice(versions, func(i, j int) bool {
+		val1, err := semver.ParseTolerant(versions[i].Version)
+		if err != nil {
+			return false
+		}
+
+		val2, err := semver.ParseTolerant(versions[j].Version)
+		if err != nil {
+			return false
+		}
+
+		return val2.LT(val1)
```

Comment:
You are sorting the version from lowest to largest. After sorted, you will find the lowest version in the `for _, templateVersion := range versions {` below.